### PR TITLE
Version v2.2.2

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

This PR contains fixes to be included in `v2.2.2`, cherry-picked from `master`; let me know if I missed any.

Commits related to release itself:
- https://github.com/DisnakeDev/disnake/pull/135/commits/2dc95d2cf35b74c5b46272424190be03a0bc1530
- https://github.com/DisnakeDev/disnake/pull/135/commits/8896bc4c0f0ec75d74f246111e73116852ea79f4

## Checklist

- [x] If code changes were made then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `black .`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/ext/commands/params.py [hunk 1]
```diff
@@ -284,7 +281,7 @@ def parse_annotation(self, annotation: Any) -> None:  # sourcery no-metrics
         if self.converter is not None:
             # try to parse the converter's annotation, fall back on the annotation itself
             parameters = list(inspect.signature(self.converter).parameters.values())
-            parameter = parameters[2] if inspect.ismethod(self.converter) else parameters[1]
+            parameter = parameters[2] if parameters[0].name == "self" else parameters[1]
```
Comment (EQUENOS) [inline disnake/ext/commands/params.py:284]: This line doesn't solve the problem in 100% cases

File: docs/api.rst [hunk 2]
```diff
@@ -4225,22 +4225,6 @@ PartialMessage
...
-    :members:
-
-OptionChoice
-~~~~~~~~~~~~
-
-.. attributetable:: OptionChoice
-
-.. autoclass:: OptionChoice()
-    :members:
-
```
Comment (EQUENOS) [inline docs/api.rst:4243]: Are these objects documented twice?

**Review by EQUENOS — COMMENTED** [review]
I haven't fixed anything critical since `2.2.1` except `from_dict` methods, so I can't really tell if we missed something important. In conclusion, I think this list of fixes is good enough. Let's wait for other opinions.

**Comment by shiftinv** [comment]
3a68767d9671db452cfb4d87f93d4ca04f17789e is also fairly important, since the examples are already updated for it but they don't actually work that way in the latest release yet.

**Comment by EQUENOS** [comment]
Isn't it included?

**Comment by shiftinv** [comment]
> Isn't it included?

Ye, it is, I just mentioned it since it's one of the two main fixes in this list. The others are primarily docs or other minor fixes.

**Review by thesadru — APPROVED** [review]

**Pull request merged**